packages/holocron/src/promiseStore.jsx

```diff
@@ -0,0 +1,87 @@
+import { useDispatch } from 'react-redux';
+
+/** @typedef {import('redux-thunk').ThunkAction} ThunkAction */
+
+/**
+ * A class to store promises by domain and key
+ *
+ * In general, Module code should not interact with the promise store directly,
+ *   instead dispatching the below thunks, or using the provided hooks
+ */
+class PromiseStore {
```

Comment:
Can we add a delete?

Reply:
I cant see a reason to delete a promise.

The purpose is to share promises across modules a LoadModuleData time.

At what point could you delete one of these promises? a single loadModuleData function does not know where it is executing, it does not know if the promise its just about to delete is still needed by some module that is going to be 'composed' in after the current moment.

So i cant think of a usecase where deleting a promise would be desirable

Reply:
They are scoped to the request. So in the server when the request finishes it will be garbage collected.

In the browser, it will only grow as much `loadModuleData` is called, and only grow as fast as the fetchye cache grows.